[자료구조] 1주차 자료구조 기능 구현

os94 opened the pull request:
안녕하세요, 저희는 **자★료★구★조** 입니다!
인스타그램을 개발하기로 결정하였고, 이번주 목표는 유저&게시글 기능구현이였습니다.
유저부분은 현재 회원가입,로그인,회원정보 수정이 가능하며,
게시글부분은 게시글 작성,조회가 가능합니다. 이때, **사진과 영상**도 첨부가능합니다.
부족한 기능 및 테스트 계속해서 추가하겠습니다.

* 저희가 파일을 프로젝트 외부폴더에 저장하고있어서,
application.properties의 로컬경로를 각 컴퓨터환경에 맞게 세팅해야합니다.
(아직 윈도우 로컬환경에서밖에 테스트해보지못해서 안돌아갈수도있을것같습니다 😢 )

ep1stas1s committed:
chore: git ignore 생성

ep1stas1s committed:
init: project 생성

ep1stas1s committed:
feat: login page

ep1stas1s committed:
feat: signup page

ep1stas1s committed:
fix: change directory

ep1stas1s committed:
fix: change directory

imkimheejoo committed:
feat: account-edit page

indiveloper committed:
add myFeed front page

imkimheejoo committed:
feat: account-edit page

imkimheejoo committed:
fix: user-edit page name fix

jinwookh committed:
add: article-edit file

imkimheejoo committed:
Merge remote-tracking branch 'origin/article_create_page' into develop

imkimheejoo committed:
Merge remote-tracking branch 'origin/feature/login-signup-page' into develop

imkimheejoo committed:
merge: myFeed page

imkimheejoo committed:
feat: 회원가입

os94 committed:
delete: 사용하지않는 html 삭제

- delete article-config, article-header, article-layout

os94 committed:
add: index.css

os94 committed:
feat: 파일 업로드 기능 샘플 추가

ep1stas1s committed:
test: user api controller test

ep1stas1s committed:
fix: html 파일 경로 수정

ep1stas1s committed:
refactor: user js 정리

ep1stas1s committed:
chore: webflux, junit5 추가

- junit4 제거

ep1stas1s committed:
chore: update git ignore

ep1stas1s committed:
feat: 제약사항 추가

ep1stas1s committed:
fix: status code 변경

ep1stas1s committed:
fix: 오류 수정

jinwookh committed:
feat: add form infos at article-edit.html

os94 committed:
feat: 메인 페이지 이동 기능 추가

os94 committed:
feat: 게시글 수정 페이지 이동 기능 구현

ep1stas1s committed:
refactor: domain, dto 구조 변경

ep1stas1s committed:
feat: 회원 수정 기능 구현

ep1stas1s committed:
chore: show sql 설정

ep1stas1s committed:
feat: login, logout 구현

os94 committed:
delete: 사용하지않는 파일 삭제

os94 committed:
refactor: 게시글 수정 페이지 이동 기능 보완

os94 committed:
feat: 게시글 생성 기능 추가

os94 committed:
feat: 게시글 조회 기능 샘플 추가

ep1stas1s committed:
feat: login interceptor 구현

ep1stas1s committed:
feat: argument resolver 적용

- session user

ep1stas1s committed:
feat: validation

- move package

ep1stas1s committed:
feat: nick name 중복 확인

ep1stas1s committed:
refactor: uri 변경

ep1stas1s committed:
feat: 추가 validation 기능 구현

ep1stas1s committed:
style: formatting

ep1stas1s committed:
fix: path 변경

ep1stas1s committed:
refactor: merge javascript file

imkimheejoo committed:
test: userService Test 코드 구현

os94 committed:
feat: 게시글 사진,동영상 노출기능 추가

imkimheejoo committed:
feat: Custom Exception 생성

os94 committed:
feat: 게시글 작성 성공시 조회 페이지로 이동

os94 committed:
refactor: 파일 저장 기능 보완

os94 committed:
Merge remote-tracking branch 'origin/feat/user' into develop

# Conflicts:
#	build.gradle
#	src/main/resources/application.properties
#	src/main/resources/static/css/index.css
#	src/main/resources/templates/index.html

os94 committed:
merge: user & article

- 일부 테스트 실패중

os94 committed:
refactor: properties 파일 수정

os94 committed:
refactor: html layout 적용 및 중복 제거

pobiconan commented on build.gradle:
```diff
@@ -0,0 +1,45 @@
+    implementation 'org.springframework.boot:spring-boot-starter-web'
+    implementation 'commons-fileupload:commons-fileupload:1.3.1'
+    implementation 'commons-io:commons-io:2.6'
+    compile 'org.projectlombok:lombok'
```
compile 대신 implementation로 통일

soojinroh replied:
implementation으로 lombok을 빌드하지 못해 compile을 사용했었는데 `annotationProcessor`으로 수정해 사용하고 있습니다.

pobiconan commented on src/main/java/com/woowacourse/dsgram/domain/Article.java:
```diff
@@ -0,0 +1,48 @@
+    private String contents;
+
+    @Column(nullable = false)
+    private String fileName;
```
길이는 기본인 255로 사용하는 것을 의도한 것인가?

soojinroh replied:
window에서 테스트해보니 파일명이 240자를 초과할 수 없어 240자로 수정하였습니다.

pobiconan commented on src/main/java/com/woowacourse/dsgram/domain/Article.java:
```diff
@@ -0,0 +1,48 @@
+
+    public String getFilePath() {
+        return filePath;
+    }
```
도메인 객체에 기본으로 equals, hashcode, toString 메소드는 구현해 주는 것이 좋지 않을까?

soojinroh replied:
equals와 hashcode는 lombok을 이용해 추가하고, toString은 override를 해주었습니다.

pobiconan commented on src/main/java/com/woowacourse/dsgram/domain/User.java:
```diff
@@ -0,0 +1,77 @@
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+@Getter
+@EqualsAndHashCode(of = {"id"})
+public class User {
```
Article은 lombok을 사용하지 않고 있다.
도메인 객체의 통일성을 가져가는 것이 어떨까?

soojinroh replied:
toString()을 제외하고는 lombok annotation을 사용하는 것으로 컨벤션을 맞추었습니다.

pobiconan commented on src/main/java/com/woowacourse/dsgram/domain/User.java:
```diff
@@ -0,0 +1,77 @@
+        this.intro = intro;
+    }
+
+    public void update(User updatedUser, String email) {
```
수정할 때 DTO가 아닌 User로 사용한 이유는?

soojinroh replied:
entity와 dto의 의존성을 낮추기 위해 dto로 사용했습니다.

pobiconan commented on src/main/java/com/woowacourse/dsgram/domain/User.java:
```diff
@@ -0,0 +1,77 @@
+    private String intro;
+
+    @Builder
+    public User(@Email String email, @Size(min = 2, max = 10) String nickName, @Size(min = 2, max = 10) String userName, @Size(min = 4, max = 16) String password, String webSite, String intro) {
```
생성자에도 유효성 체크 관련 설정하는 것은 괜찮은가?

soojinroh replied:
이미 인스턴스 변수에서 설정을 했기 때문에 생성자에서 다시 유효성 체크 관련하여 설정할 필요가 없다고 생각하여 제거하였습니다.

pobiconan commented on src/main/java/com/woowacourse/dsgram/domain/UserRepository.java:
```diff
@@ -0,0 +1,11 @@
+public interface UserRepository extends JpaRepository<User, Long> {
+    Optional<User> findByEmail(String email);
+
+    long countByNickName(String nickName);
```
이 메소드가 필요한 이유는?
필요하다면 반환 타입이 long이어야 하는가? int면 되지 않을까?

soojinroh replied:
countByNickName()을 existsByNickName으로 수정하여 boolean을 return받아 사용하도록 개선하였습니다.

pobiconan commented on src/main/java/com/woowacourse/dsgram/service/ArticleApiService.java:
```diff
@@ -0,0 +1,28 @@
+    public Article create(Article article) {
+        try {
+            return articleRepositoy.save(article);
+        } catch (Exception exception) {
```
이 예외 처리가 의미가 있나?
이와 같이 예외 처리하는 이유는?

soojinroh replied:
Jpa exception을 별도로 처리하려고 위와 같이 try catch로 묶어주었는데
Jpa가 던지는 DataAccessException을 RestExceptionHandler에서 처리할 수 있을 것 같아 위의 로직을 해당 handler로 위임하였습니다.

pobiconan commented on src/main/java/com/woowacourse/dsgram/service/FileService.java:
```diff
@@ -0,0 +1,50 @@
+public class FileService {
+
+    @Value("${ARTICLE_UPLOAD_PATH}")
+    private String ARTICLE_UPLOAD_PATH;
```
인스턴스 변수인데 변수 이름을 대문자로 구현한 이유는?

soojinroh replied:
수정하였습니다!

soojinroh replied:
해당 변수가 인스턴스 변수처럼 보이지만 @Value annotation을 이용해 application.properties에서 전역변수처럼 쓰이며 사실상 기능적으로는 상수와 같이 사용되는 것 같은데 어떤 컨벤션을 따르는 게 좋을까요?

pobiconan commented on src/main/java/com/woowacourse/dsgram/service/FileService.java:
```diff
@@ -0,0 +1,50 @@
+    private String ARTICLE_UPLOAD_PATH;
+
+    @Value("${TOMCAT_PATH}")
+    private String TOMCAT_PATH;
```
tomcat 경로가 왜 필요한가?
tomcat 경로 없이 구현할 수 있지 않을까?

os94 replied:
파일을 저장하는 방식으로 S3보다, 로컬을 이용한 기본적인 방법을 공부해보기로 결정하고 기능구현을 진행하였습니다. 그 과정에서 별도의 path설정없이 기본루트로 저장시 톰캣의 `docBase@@@`와 같은 서버 실행시마다 달라지는 임시폴더로 저장되는 문제를 알수있었고, path를 설정하다보니 파일을 resources같은 프로젝트 폴더 내부에 저장할경우 단점이 존재한다는것을 알게되었습니다. 이후 프로젝트 외부에 저장하는 방법을 열심히 찾아봤고, 그렇게 찾은 최선의 방법이 이렇게 application.properties와 절대경로를 이용해 저장하는 방법이었습니다.
그 과정에서 저장경로에 `/work/Tomcat/localhost/ROOT`라는 톰캣path가 추가되는데, 이런 설정(?)을 다루는 방법을 아직 찾지못했습니다 😢 방법을 더 찾아보겠습니다.!

pobiconan commented on src/main/java/com/woowacourse/dsgram/service/UserService.java:
```diff
@@ -0,0 +1,69 @@
+        userRepository.save(UserAssembler.toEntity(signUpUserDto));
+    }
+
+    private void checkDuplicatedAttributes(String nickName, String email) {
```
칼럼에 unique 설정이 되어 있다면 굳이 이런 처리가 필요없지 않을까?

pobiconan commented on src/main/java/com/woowacourse/dsgram/service/dto/ArticleRequest.java:
```diff
@@ -0,0 +1,53 @@
+
+import org.springframework.web.multipart.MultipartFile;
+
+public class ArticleRequest {
```
User는 DTO이고 Article은 Request인가?
전체 소스 코드의 컨벤션을 통일한다.

soojinroh replied:
Request와 Response를 사용하는 것으로 컨벤션을 통일하였습니다.

pobiconan commented on src/main/java/com/woowacourse/dsgram/web/config/WebConfig.java:
```diff
@@ -0,0 +1,31 @@
+                .excludePathPatterns("/images/**")
+                .excludePathPatterns("/login")
+                .excludePathPatterns("/signup")
+                .excludePathPatterns("/api/users/login");
```
이런 설정 부분도 파일 경로 처리와 같이 설정 파일로 분리해 처리하는 것은 어떨까?

os94 replied:
다른 경로들과 마찬가지로 application.properties 설정으로 분리하였습니다.

pobiconan commented on src/main/resources/application.properties:
```diff
@@ -0,0 +1,17 @@
+
+ARTICLE_UPLOAD_PATH= /articles/files
+TOMCAT_PATH= /work/Tomcat/localhost/ROOT
+BASE_DIR= C:/Users/ohsang seo/Desktop/public
```
이와 같이 구현할 경우 aws 개발 서버에 어떻게 배포할 것인가?

os94 replied:
Window뿐만아니라, Max과 Linux 환경에서도 각각에 알맞는 적절한 path를 설정할수있도록, 주석으로 3가지 path를 준비해두었습니다.
추후, `System.getProperty()`를 이용해 일관된 방법으로 관리할수있도록 리팩토링할 예정입니다!

pobiconan reviewed (changes requested):
Article과 User 기능이 같은 컨벤션으로 구현되어 있지 않네요.
시작 단계에서 기능을 빨리 추가하는 것도 좋지만 그 보다 컨벤션을 통일하는 것이 더 중요합니다.
컨벤션 통일해 주세요.

wmakerjun commented on src/main/resources/templates/index.html:
```diff
@@ -0,0 +1,183 @@
+    <title>woostagram</title>
+</th:block>
+<th:block layout:fragment="content">
+        <div class="page-container mrg-top-120">
```
html코드도 컨벤션을 맞춰주는게 좋습니다. IDE의 folding 기능을 이용해서 컨벤션이 맞는지 확인 후 같이 잡아주면 좋을 것 같아요 :) 
현재는 아래와 같이 보이네요~! 
<img width="619" alt="스크린샷 2019-08-16 오전 10 37 22" src="https://user-images.githubusercontent.com/50367798/63137585-d90d1100-c011-11e9-9a98-19a4c917681b.png">

soojinroh replied:
컨벤션 수정하였습니다!

wmakerjun commented on src/main/resources/templates/signup.html:
```diff
@@ -0,0 +1,99 @@
+                                ―――――――― 또는 ――――――――
+                            </div>
+                            <div>
+                                <form class="form-group mrg-vertical-15">
```
현재 이메일을 작성하지 않아도 회원가입이 되는 오류가 발생하고 있습니다. 당연히 서버에서도 유효성을 검증해야하지만, 그전에 사용자가 잘못된 입력값을 서버에 보내지 않도록 프론트에서 1차적으로 처리를 해주면 더 좋습니다.
HTML5에서는 Form에서 간단하게 반드시 전송해야 하는 데이터를 강제하게 해주는 옵션이 있습니다.

[링크](https://www.w3schools.com/html/html_form_attributes.asp)를 참조해서 프론트에서도 유효성을 체크할 수 있게 해주세요 :)

soojinroh replied:
input tag에 required와 min_length, max_length 속성을 추가하였습니다!

wmakerjun commented on src/main/resources/templates/article-edit.html:
```diff
@@ -0,0 +1,42 @@
+    </div>
+</th:block>
+
+<script src="/js/article-edit.js"></script>
```
현재 article-edit.html 페이지에서 피드를 작성하고 저장버튼을 눌렀을 때 아무런 동작이 되지 않습니다.
네트워크탭을 보면 article-edit.js가 load되지 않았음을 확인할 수 있는데요. 원인을 한번 찾아 보면 타임리프를 이해하는데 더욱 도움이 될 것 같아요 :)

**hint**: fragments와 layout.html

soojinroh replied:
반영했습니다!

wmakerjun commented on src/main/resources/templates/article-edit.html:
```diff
@@ -0,0 +1,42 @@
+                <div class="row">
+                    <div class="col-lg-7 mx-auto">
+                        <div class="card widget-feed no-pdd mrg-btm-70">
+                            <form method="post" action="/articles" id="saveForm" name="saveForm"
```
29line에 있는 버튼의 id는 save-button 인데, form에 있는 id는 saveForm으로 네이밍컨벤션이 다릅니다. 다른 클래스와도 일치할 수 있게 save-form으로 통일해주면 더 좋을 것 같아요 :)

soojinroh replied:
컨벤션 통일했습니다!

wmakerjun commented on src/main/resources/static/js/user.js:
```diff
@@ -0,0 +1,152 @@
+                headers: {'Content-Type': 'application/json; charset=UTF-8'},
+                body: JSON.stringify(userBasicInfo)
+            })
+                .then(response => {
```
fetch와 then을 아래처럼 컨벤션을 맞춰주면 더 좋습니다 :)
<img width="645" alt="스크린샷 2019-08-16 오전 11 07 01" src="https://user-images.githubusercontent.com/50367798/63138586-05c32780-c016-11e9-910f-50c7d8948812.png">

soojinroh replied:
컨벤션 수정했습니다!

wmakerjun commented on src/main/resources/static/js/user.js:
```diff
@@ -0,0 +1,152 @@
+                password: password.value,
+            };
+
+            fetch('/api/users/login', {
```
fetch api를 이용할 때 같은 형식이 반복되면서 중복코드(ex_ headers: {'Content-Type': 'application/json; charset=UTF-8'},)가 많이 발생합니다. 또한 앞으로 api가 늘어날 때마다 모든 api 요청을 매번 작성하려면 빠진 api는 없는지, api간에 중복된 것은 없는지 체크하기가 어렵습니다. UserController와 UserService를 만든것처럼 Api를 다루는 부분을 따로 만들어서 이용하면 중복된 코드를 간결하게 하면서 훨씬 가독성 좋게 만들 수 있을 수 있습니다.

```javascript
const Api = function () {
        const request = (url, method, body) => {
            const defaultHeader = {'Content-Type': 'application/json; charset=UTF-8'}
            return fetch(url, {
                method: method,
                headers: defaultHeader,
                body: body
            })
        }

        const createUser = (userBasicInfo) => {
            return request('/api/users', 'post', userBasicInfo)
        }

        return {
            createUser: createUser
        }
    }

  ...
  
  const api = new Api()
  
  api
    .createUser(JSON.stringify(userBasicInfo))
    .then(response => {
        if (response.status === 200) {
            window.location.href = '/login';
        }
        if (response.status === 400) {
            throw response;
        }
    })
    .catch(error => {
        error.json()
            .then(exception => {
                alert(exception.message)
            });
    });
```

os94 replied:
fetch template을 만들어 분리해보았습니다!

wmakerjun commented on src/main/resources/static/js/user.js:
```diff
@@ -0,0 +1,152 @@
+                    throw response;
+                }
+            }).catch(error => {
+                error.json()
```
error 처리하는 코드가 반복되고 있습니다. errorHandler 함수를 따로 만들어서 처리해주면 어떨까요~?

soojinroh replied:
반영했습니다!

wmakerjun commented on src/main/resources/application.properties:
```diff
@@ -0,0 +1,17 @@
+
+ARTICLE_UPLOAD_PATH= /articles/files
+TOMCAT_PATH= /work/Tomcat/localhost/ROOT
+BASE_DIR= C:/Users/ohsang seo/Desktop/public
```
[링크](https://stackoverflow.com/questions/30120813/how-do-i-provide-a-file-path-in-mac-os-x-while-creating-a-file-in-java) 참조해서 겨우 테스트해보았네요 ㅎㅎ.... 다른 방법을 찾아서 적용해보면 좋을 것 같아요~!

os94 replied:
Window뿐만아니라, Max과 Linux 환경에서도 각각에 알맞는 적절한 path를 설정할수있도록, 주석으로 3가지 path를 준비해두었습니다.
추후, System.getProperty()를 이용해 일관된 방법으로 관리할수있도록 리팩토링할 예정입니다!

wmakerjun commented on src/main/resources/templates/fragments/header.html:
```diff
@@ -0,0 +1,50 @@
+                </div>
+            </li>
+            <li class="search-box mrg-left-20 desktop-view">
+                <a class="search-toggle no-pdd-right" href=";">
```
href에 의미없는 세미콜론은 삭제해주는게 좋을 것 같아요~!

soojinroh replied:
삭제했습니다!

wmakerjun reviewed (changes requested):
프론트엔드 부분이 많이 추가되서 깜짝놀랐어요 ㅎㅎ 큰 부분들은 구조도 잘 잡고 잘 진행되고 있는것 같아요. 
너무 훌륭합니다 👏😃
몇몇 부분만 수정하면 앞으로 구현하는 프론트 코드품질이 훨씬 향상될 것 같아요~!

ep1stas1s committed:
refactor: exclude path pattern 설정 파일로 분리

ep1stas1s committed:
feat: github oauth 적용

ep1stas1s committed:
fix: gradle 수정

- reformat

ep1stas1s committed:
refacotr: js 중복제거

imkimheejoo committed:
refactor: 회원정보 수정 Test 리팩토링

indiveloper committed:
fix: html파일 js 로드 실패 수정
1. article-edit.html에서 article-edit.js 가져올 수 있도록 수정
2. article.html에서 js 코드 적용 안 되던 것 수정

imkimheejoo committed:
refactor: UserApiController의 공통로직 부모클래스로 분리및 테스트 추가

ep1stas1s committed:
feat: 로그인한 사용자에 대한 interceptor 구현

- 기존 interceptor name 수정

ep1stas1s committed:
fix: redirect 경로 변경

ep1stas1s committed:
refactor: exception package 분리

imkimheejoo committed:
fix: email validate 수정

ep1stas1s committed:
fix: name이 null일 경우 처리

ep1stas1s committed:
fix: 제약조건 추가

ep1stas1s committed:
fix: email 제약조건 변경

ep1stas1s committed:
fix: 오작동하는 method 수정

ep1stas1s committed:
refactor: 중복 제거

ep1stas1s committed:
chore: interceptor 항목 추가 (in test properties)

ep1stas1s committed:
refactor: http connection 중복 제거

woowahanCU commented on build.gradle:
```diff
@@ -0,0 +1,45 @@
+    mavenCentral()
+}
+
+dependencies {
```
현재 로컬에서 테스트시에 실패하고 있네요. 
그런데 gradlew 로 build 실행시에는 테스트를 skip하고 있어요. 
아래 설정 추가한 이후에 배포해보시겠어요?
```java
test {
    useJUnitPlatform()
}
...
dependencies {
...
    testImplementation 'org.junit.jupiter:junit-jupiter-api:5.3.1'
    testRuntimeOnly 'org.junit.jupiter:junit-jupiter-engine:5.3.1'
```java

https://www.baeldung.com/junit-5-gradle

woowahanCU commented on src/main/java/com/woowacourse/dsgram/service/ArticleApiService.java:
```diff
@@ -0,0 +1,28 @@
+
+    @Transactional(readOnly = true)
+    public Article findById(long articleId) {
+        return articleRepositoy.findById(articleId).orElseThrow(() -> new JpaException("해당 게시글을 조회하지 못했습니다."));
```
Exception은 표준 예외가 있을 경우에 이를 활용해주는 것이 좋습니다. 그리고 Exception에 상세한 정보를 많이 담는 것이 좋으며 추상화 수준에 맞는 예외를 던지라고 Effective java에서 가이드하고 있어요.
https://jaehun2841.github.io/2019/03/10/effective-java-item72/#unsupportedoperationexception

이 경우 JPA의 Exception 중 EntityNotFoundException을 활용해도 좋지 않을까요. 
그리고 어떤 게시글인지에 대한 정보도 같이 넘겨주어도 좋겠네요. 

https://www.objectdb.com/api/java/jpa/exceptions

soojinroh replied:
JpaException에서 EntityNotFoundException으로 수정하였습니다.

woowahanCU commented on src/main/resources/application.properties:
```diff
@@ -0,0 +1,17 @@
+
+ARTICLE_UPLOAD_PATH= /articles/files
+TOMCAT_PATH= /work/Tomcat/localhost/ROOT
+BASE_DIR= C:/Users/ohsang seo/Desktop/public
```
각 설정들이 팀원들이 로컬에서 테스트할 경우에도 서로간의 공유가 되어야 할 거 같은데요.
현재 어떤 방식으로 정보를 관리하고 있나요?

os94 replied:
Window뿐만아니라, Max과 Linux 환경에서도 각각에 알맞는 적절한 path를 설정할수있도록, 주석으로 3가지 path를 준비해두었습니다.
추후, System.getProperty()를 이용해 일관된 방법으로 관리할수있도록 리팩토링할 예정입니다!

woowahanCU commented on src/main/java/com/woowacourse/dsgram/web/controller/ArticleApiController.java:
```diff
@@ -0,0 +1,38 @@
+    public ResponseEntity<Article> create(ArticleRequest articleRequest) {
+        FileInfo fileInfo = fileService.save(articleRequest.getFile());
+        Article article = convertFrom(articleRequest, fileInfo);
+        Article savedArticle = articleApiService.create(article);
```
현재는 ArticleService를 별도로 구성한 이점이 없어보이네요.

woowahanCU commented on src/main/java/com/woowacourse/dsgram/web/controller/UserApiController.java:
```diff
@@ -0,0 +1,56 @@
+import javax.validation.Valid;
+
+@RestController
+@RequestMapping("/api/users")
```
ArticleApiController와 url path 구조가 다르게 구성한 이유가 있을까요?

soojinroh replied:
통일시켰습니다!

woowahanCU reviewed (changes requested):
테스트하려면 어떻게 해야 하나요? 🤔
별도의 공유문서가 안보이는데요. 어떻게 공유하고 있나요?

imkimheejoo committed:
merge: userTest,email이슈 merge

imkimheejoo committed:
fix: signup, account-edit 프론트 유효성 검사

imkimheejoo committed:
fix: front load,naming feedback 수정

imkimheejoo committed:
refactor: error 중복제거

indiveloper committed:
test: ArticleApiServiceTest 코드 추가

1. 게시글 작성 성공 테스트 코드 추가
2. 게시글 조회 성공 테스트 코드 추가

각각의 실패 테스트 코드 추가 필요

indiveloper committed:
refactor: 로컬의 파일을 읽어오는 기능을 ArticleController에서 FileService로 분리 및 이동

ArticleController에서 로컬의 파일을 읽어와 byte[] base64로 변환하는 기능을 FileService의 readFile() 메소드로 분리 구현

indiveloper committed:
refactor: RestFul한 showArticleFile()메서드를 ArticleController에서 ArticleApiController로 이동

indiveloper committed:
refactor: ArticleApiService의 DI주입 방식 변경 및 JPA exception 발생 메서드 수정

1. 필드 주입으로 DI 받던 ArticleRepository를 생성자 주입 방식으로 변경
2. create() 메소드에서 Exception을 catch해서 JPA Exception을 handling하던 것에서 DataAccessException을 catch하는 것으로 수정

indiveloper committed:
Merge branch 'feat/article-test' into develop

# Conflicts:
#	src/main/java/com/woowacourse/dsgram/web/controller/ArticleApiController.java
#	src/main/java/com/woowacourse/dsgram/web/controller/ArticleController.java

indiveloper committed:
refactor: Article Lombok getter() 적용 및 toString, hashcode, equals 구현

indiveloper committed:
refactor: 변수명 컨벤션 수정

1. 인스턴스 변수를 상수 컨벤션으로 작성하던 것을 캐멀케이스로 수정

indiveloper committed:
feat: multipartfile size limit 설정 추가

indiveloper committed:
refactor: ArticleRequest에 Lombok 적용

indiveloper committed:
refactor: class명 변경 및 사용하지 않는 파일 삭제

1. DTO 이름 컨벤션 통일
2. 사용하지 않는 empty.txt 파일 삭제

indiveloper committed:
refactor: Article의 fileName length 설정 추가

1. fileName의 length를 240로 제한

indiveloper committed:
refactor: User에 toString() 추가

indiveloper committed:
refactor: 생성자에서 valid annotation 제거

indiveloper committed:
refactor: 컨벤션 수정 및 파일명 변경

indiveloper committed:
refactor: UserRepository의 countByNickName을 existsByNickName으로 수정

indiveloper committed:
refactor: jpa exception handling 중복 제거

indiveloper committed:
refactor: WebConfig 경로처리를 설정파일로 분리

indiveloper committed:
Merge remote-tracking branch 'origin/fix/frontweb' into develop

# Conflicts:
#	src/main/resources/templates/article-edit.html
#	src/main/resources/templates/article.html

indiveloper committed:
style: 컨벤션 수정

indiveloper committed:
refactor: ArticleApiService findById() exception 수정

JpaException을 EntityNotFoundException으로 수정

indiveloper committed:
refactor: ArticleApiController url path 수정

/articles -> /api/articles

indiveloper committed:
Merge remote-tracking branch 'origin/feat/oauth' into develop

# Conflicts:
#	src/main/java/com/woowacourse/dsgram/service/UserService.java
#	src/main/java/com/woowacourse/dsgram/service/dto/user/SignUpUserRequest.java
#	src/main/java/com/woowacourse/dsgram/web/config/WebConfig.java
#	src/main/resources/application.properties
#	src/main/resources/static/js/user.js
#	src/test/resources/application.properties

indiveloper committed:
merge: merge with feat/oauth

indiveloper committed:
chore: TODO 주석 정리

os94 committed:
fix: 게시글 파일 로딩 버그 수정

jinwookh committed:
feat: add home link to logo icon

imkimheejoo committed:
test: userService 실패 메서드 수정

ep1stas1s committed:
Merge remote-tracking branch 'origin/fix/usertest2' into develop

jinwookh committed:
test: 테스트 모두 성공

wmakerjun commented on src/main/resources/static/js/article-edit.js:
```diff
@@ -0,0 +1,30 @@
+const saveButton = document.getElementById("save-button");
+
+const save = function () {
+    let file = document.getElementById("file").value;
```
file의 값을 변경하지 않는데 따로 let을 쓴 이유가 있을까요?

wmakerjun commented on src/main/resources/templates/fragments/script.html:
```diff
@@ -0,0 +1,26 @@
+                document.querySelector(".advanced-search").classList.add("active")
+            } else {
+                document.querySelector(".advanced-search").classList.remove("active")
+                $('.advanced-search').removeClass("active");
```
이건 제 실수인데요 😭 😭 😭 해당 22 line은 21 line과 똑같은 기능을 하는 코드입니다. 그래서 삭제 부탁드릴게요 😭 😭 😭

wmakerjun commented on src/main/resources/static/js/article-edit.js:
```diff
@@ -0,0 +1,30 @@
+    let postForm = document.getElementById("saveForm");
+    let formData  = new FormData(postForm);
+
+    fetch ('/articles', {
```
user.js에서 api 역할을 하는 객체를 따로 만들었던 것처럼 여기서도 분리해보면 어떨까요~?

wmakerjun commented on src/main/resources/static/js/article-edit.js:
```diff
@@ -0,0 +1,30 @@
+
+
+const saveButton = document.getElementById("save-button");
```
이런식으로 변수를 선언하는 것은 **전역변수**로 선언하는 것입니다. saveButton이라는 네이밍이 나중에 다른 기능이나 라이브러리의 변수명과 충돌이 날 수 있기 때문에, articeSaveButton과 같이 특정지어주거나 scope를 만들어주면 좋을 것 같아요~! 
아래 링크들을 참조해주시면 좋을 것 같습니다 :)
https://webclub.tistory.com/312
https://devyj.tistory.com/9

wmakerjun reviewed (changes requested):
빠르게 수정할 수 있는 몇가지 추가적인 요구사항을 남겼습니다 해당건 해결되는대로 바로 approve 진행할게요 😄
